<?xml version="1.0" encoding="UTF-8"?>
<root BTCPP_format="4"
      main_tree_to_execute="BehaviorTree">
  <BehaviorTree ID="BehaviorTree">
    <ReactiveSequence>
      <SentryFollower topic="/goal_pose"
              follow_distance="2.0"
              current_goal="{current_goal}"/>
      <!-- TODO: set goal_x and goal_y -->
      <SendGoal goal_x="0.0" goal_y="0.0"/>
    </ReactiveSequence>
  </BehaviorTree>

  <!-- Description of Node Models (used by Groot) -->
  <TreeNodesModel>
    <Action ID="SendGoal">
      <input_port name="goal_pose"
                  default="{goal_pose}"/>
      <input_port name="goal_x" default="0.0"/>
      <input_port name="goal_y" default="0.0"/>
    </Action>
    <Action ID="SentryFollower">
      <input_port name="topic"
                  default="/goal_pose"/>
      <input_port name="follow_distance"
                  default="2.0"/>
      <output_port name="current_goal"
                   default="{current_goal}"/>
    </Action>
  </TreeNodesModel>

</root>
